build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -1600,6 +1662,78 @@ message BatchUpdateBlobsResponse {
   repeated Response responses = 1;
 }
 
+message ConcatenateBlobsRequest {
+  message Request {
+    // The digest of the blob to create.
+    Digest digest = 1;
+
+    // For BLAKE3CONCAT, the message words m0...m15 of the root parent
+    // nodes of the smaller blobs, stored in little-endian order.
+    //
+    // The first 64*n bytes correspond to the root nodes of n blobs
+    // having size 2^k, where k > 10. The final 64 bytes correspond to
+    // the root node of the blob containing trailing data, containing l
+    // bytes of data, where 0 < l <= 2^k. The server may infer the
+    // values of k and l using `digest.size_bytes` and the length of
+    // this field.
+    bytes chaining_values = 2;
+  }
+
+  // The instance of the execution system to operate against. A server may
+  // support multiple instances of the execution system (with their own workers,
+  // storage, caches, etc.). The server MAY require use of this field to select
+  // between them in an implementation-defined fashion, otherwise it can be
+  // omitted.
+  string instance_name = 1;
+
+  // The digest function used for constructing the concatenated blobs.
+  // This field MUST have value BLAKE3CONCAT, as it is the only supported
+  // digest function that uses a Merkle tree construction.
+  DigestFunction.Value digest_function = 2;
+
+  // The individual concatenation requests.
+  repeated Request requests = 3;
+}
+
+message SplitBlobsRequest {
+  // The instance of the execution system to operate against. A server may
+  // support multiple instances of the execution system (with their own workers,
+  // storage, caches, etc.). The server MAY require use of this field to select
+  // between them in an implementation-defined fashion, otherwise it can be
+  // omitted.
+  string instance_name = 1;
+
+  // The digest function used for constructing the split blobs. This
+  // field MUST have value BLAKE3CONCAT, as it is the only supported
+  // digest function that uses a Merkle tree construction.
+  DigestFunction.Value digest_function = 2;
+
+  // The digests of the large blobs to split into smaller blobs.
+  repeated Digest digests = 3;
+
+  // The size of the non-trailing blobs to create. It must be less than
+  // the sizes stored in `digests`. Furthermore, for BLAKE3CONCAT must
+  // be a power of 2.
+  int64 split_size_bytes = 4;
```

Comment:
It depends. If the client wants really granular access to the object with as much deduplication as possible, it can use `blake3concat_min_split_size_bytes`. If it simply wants to download the object in its entirety without caring too much about about deduplication, it can use the highest power of two below `max_batch_total_size`.